### PR TITLE
IDEM-179

### PR DESCRIPTION
**Pull request by pondpiu**
Implement #179 

- add optional field `requiredClaimNnemonics` to sites.json
- disable signup button based on whether user has claims satisfying vendor's `requiredClaimNnemonics` constrain or not

### CODE REVIEW

File: app/src/navigation/screens/VendorDetails.tsx [hunk 1]
```diff
@@ -46,6 +46,26 @@ const VendorDetailsScreen: React.FC = () => {
   const vendor = vendors.find((v) => v.name === route.params.vendorId);
   const [signed, setSigned] = React.useState<boolean>(false);
 
+  const hasAllRequiredClaims = React.useMemo(() => {
+    if (!vendor || !vendor.requiredClaimNnemonics) {
+      return true;
+    }
+
+    const userClaimNnemonicMap = usersClaims.reduce(
```
Comment (bitcoinbrisbane) [inline app/src/navigation/screens/VendorDetails.tsx:54]: Spelling error

```suggestion
    const userClaimMnemonicMap = usersClaims.reduce(
```

File: app/src/navigation/screens/VendorDetails.tsx [hunk 2]
```diff
@@ -106,21 +126,26 @@ const VendorDetailsScreen: React.FC = () => {
...
               }
               if (name && email) {
-                makeUser.signUp(name, email);
+                await makeUser.signUp(name, email);
+                setSigned(true);
+              } else {
+                Alert.alert(
+                  "Missing Credentials",
+                  "Please provide your name and email claims to sign up"
+                );
```
Comment (bitcoinbrisbane) [inline app/src/navigation/screens/VendorDetails.tsx:143]: Is this possible to get to given we disable the button?  If not, lets just remove

File: app/src/types/general.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ export type Vendor = {
   tagline: string;
   website: string;
   backgroundColor: string;
+  requiredClaimNnemonics: string[];
```
Comment (bitcoinbrisbane) [inline app/src/types/general.ts:18]:
```suggestion
  requiredClaimMnemonics: string[];
```

File: data/sites.json [hunk 4]
```diff
@@ -8,7 +8,8 @@
     "logo": "https://images.squarespace-cdn.com/content/v1/5f5990cc9e792617828214ae/1600921877148-3ISFDDENM1N89GLGAIKI/GBIB+Logo_5%402x.png?format=1500w",
     "signup": "https://portal.getpaidinbitcoin.com.au/register",
     "useProxy": false,
-    "backgroundColor": "#be2596"
+    "backgroundColor": "#be2596",
+    "requiredClaimNnemonics": ["email", "mobile"]
```
Comment (bitcoinbrisbane) [inline data/sites.json:12]:
```suggestion
    "requiredClaimMnemonics": ["email", "mobile"]
```

File: data/sites.json [hunk 5]
```diff
@@ -18,7 +19,8 @@
     "logo": "https://res.cloudinary.com/dcisquypm/image/upload/v1651057203/validator-logos/ykib0ji0gekmjfcjhp48.png",
     "signup": "https://coinstash.com.au/auth/signup",
     "useProxy": false,
-    "backgroundColor": "#0012d0"
+    "backgroundColor": "#0012d0",
+    "requiredClaimNnemonics": ["email", "mobile"]
```
Comment (bitcoinbrisbane) [inline data/sites.json:23]:
```suggestion
    "requiredClaimMnemonics": ["email", "mobile"]
```

File: data/sites.json [hunk 6]
```diff
@@ -29,7 +31,8 @@
     "logo": "https://pngset.com/images/dogecoin-wow-gold-gold-medal-trophy-money-transparent-png-1137773.png",
     "signup": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
     "useProxy": false,
-    "backgroundColor": "blue"
+    "backgroundColor": "blue",
+    "requiredClaimNnemonics": ["email", "mobile", "eighteen"]
```
Comment (bitcoinbrisbane) [inline data/sites.json:35]:
```suggestion
    "requiredClaimMnemonics": ["email", "mobile", "eighteen"]
```

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Few issues

File: app/src/navigation/screens/VendorDetails.tsx (hunk 2 above)
Comment (pondpiu) [reply app/src/navigation/screens/VendorDetails.tsx:143]: depends on whether the vendor's required claims contain name & email or not
ie. vendor that doesn't required any claims would still trigger that alert

File: app/src/navigation/screens/VendorDetails.tsx [hunk 7]
```diff
@@ -46,6 +46,26 @@ const VendorDetailsScreen: React.FC = () => {
...
   const [signed, setSigned] = React.useState<boolean>(false);
 
+  const hasAllRequiredClaims = React.useMemo(() => {
+    if (!vendor || !vendor.requiredClaimMnemonics) {
+      return true;
+    }
+
+    const userClaimMnemonicMap = usersClaims.reduce(
+      (acc, claim) => {
+        acc[claim.nnemonic] = true;
```
Comment (xiangzhe0812) [inline app/src/navigation/screens/VendorDetails.tsx:56]: mnemonic?

**Review by amagon270 — APPROVED** [review]

**Review by JasminFarhi — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Pull request merged**